src/components/Map.js

```diff
@@ -29,9 +35,9 @@ class Map extends Component {
     });
   }
 
-  shouldComponentUpdate(nextProps) {
+  componentDidUpdate(nextProps) {
+    let map = this.state.map;
```

Comment:
Maybe this should be const instead of let? I do not see where it's reassigned at all.